### PR TITLE
Camel 3.14 - Whats New Blog. WIP

### PR DESCRIPTION
**Pull request by davsclaus**

### CODE REVIEW

**Comment by davsclaus** [comment]
Need to complete the kafka section

Feedback is welcome, and also fix any grammar and typos directly please.

File: content/blog/2021/12/camel314-whatsnew/index.md [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+### Better Java 17 support
+
+Although Java 17 is not officially supported, we did improve a few Camel components to make them work with Java 17.
+The official support is Java 11 (primary) and Java 8 (secondary).
+
+### New components
+
+This release has a number of new components:
+
+- `camel-azure-servicebus` - Stream events from or to Azure Event Hubs using AMQP protocol.
```
Comment (oscerd) [inline content/blog/2021/12/camel314-whatsnew/index.md:120]: I think this should be removed. It has been introduced in 3.12.0 as far as I remember

File: content/blog/2021/12/camel314-whatsnew/index.md [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+
+## Upgrading
+
+Make sure to read the [upgrade guide](/manual/camel-3x-upgrade-guide-3_14.html) if you are upgrading from a previous Camel version.
+
+## Release Notes
+
+You can find more information about this release in the with a list of JIRA tickets resolved in the release: 
+
+- [release notes 3.12](/releases/release-3.12.0/)
```
Comment (oscerd) [inline content/blog/2021/12/camel314-whatsnew/index.md:135]: I guess we only need 3.14.0

**Review by oscerd — APPROVED** [review]
Just two minor findings

**Comment by davsclaus** [comment]
Oh this is a blot post for the last 6 months, eg it includes 3.12, 3.13 and 3.14 all together - since the last LTS release

**Comment by oscerd** [comment]
Ah silly me, I was milseaded from the PR title. Sorry my mystake.

File: content/blog/2021/12/camel314-whatsnew/index.md [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+---
+
+Apache Camel 3.14 has just been [released](/blog/2021/12/RELEASE-3.14.0/).
+
+This is the **last** LTS release supporting Java 8, and therefore we have
+extended the support period from 1 to 2 years.
+
+This blog post first details the noteworthy changes since the last 3.11 LTS release from 6 months ago.
+
+## So what's in this release (6 months for work)
```
Comment (gzurowski) [inline content/blog/2021/12/camel314-whatsnew/index.md:16]:
```suggestion
## So what's in this release (6 months of work) 
```

File: content/blog/2021/12/camel314-whatsnew/index.md [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+
+This blog post first details the noteworthy changes since the last 3.11 LTS release from 6 months ago.
+
+## So what's in this release (6 months for work) 
+
+This release introduces a set of new features and noticeable improvements that we will cover in this blog post.
+
+### Documentation
+
+We have polished, cleaned up, and updated most of the documentation in 2nd half this year.
```
Comment (gzurowski) [inline content/blog/2021/12/camel314-whatsnew/index.md:22]:
```suggestion
We have polished, cleaned up, and updated most of the documentation in the 2nd half this year.
```

File: content/blog/2021/12/camel314-whatsnew/index.md [hunk 5]
```diff
@@ -0,0 +1,141 @@
...
+We will continue this work in 1st half next year to complete this work.
```
Comment (gzurowski) [inline content/blog/2021/12/camel314-whatsnew/index.md:23]:
```suggestion
We will continue this work in the 1st half next year to complete this work.
```

File: content/blog/2021/12/camel314-whatsnew/index.md [hunk 6]
```diff
@@ -0,0 +1,141 @@
...
+
+### Core
+
+Configuring data formats, and thread pools are now reflection free (Native compilation friendly).
```
Comment (gzurowski) [inline content/blog/2021/12/camel314-whatsnew/index.md:27]:
```suggestion
Configuring data formats and thread pools are now reflection free (native compilation friendly).
```

File: content/blog/2021/12/camel314-whatsnew/index.md [hunk 7]
```diff
@@ -0,0 +1,141 @@
...
+
+We identified a few places to improve performance by reducing object allocations
+in the Camel event notification system.
+
+Logging with logMask=true to hide sensitive information now masks all known
```
Comment (gzurowski) [inline content/blog/2021/12/camel314-whatsnew/index.md:32]:
```suggestion
Logging with `logMask=true` to hide sensitive information now masks all known
```

File: content/blog/2021/12/camel314-whatsnew/index.md [hunk 8]
```diff
@@ -0,0 +1,141 @@
...
+secured keywords gather from all the Camel components.
```
Comment (gzurowski) [inline content/blog/2021/12/camel314-whatsnew/index.md:33]:
```suggestion
secured keywords gathered from all the Camel components.
```

File: content/blog/2021/12/camel314-whatsnew/index.md [hunk 9]
```diff
@@ -0,0 +1,141 @@
...
+The tracer can now be set to standby, which allows enabling tracing later during runtime.
+
+### Route Configuration
+
+Route configuration is used for separating configurations from the routes. This can be used in situations such as configuring different error handling across a set of routes. In previous versions of Camel this was more cumbersome to do, as you would either have to copy the same configuration to a set of routes or rely on global error handling configuration.
+Now you can configure a number of route configurations, and then specify on each route which configuration to use (you can use match by ids, wildcards, and regular expression).
+
+### Reload Routes
+
+The route reload functionality in Camel is capable of watching a directory folder for file changes, and then automatic trigger reload of the running routes in the Camel application.
```
Comment (gzurowski) [inline content/blog/2021/12/camel314-whatsnew/index.md:46]:
```suggestion
The route reload functionality in Camel is capable of watching a directory folder for file changes, and then automatically trigger a reload of the running routes in the Camel application.
```

File: content/blog/2021/12/camel314-whatsnew/index.md [hunk 10]
```diff
@@ -0,0 +1,141 @@
...
+
+A Kamelet is now capable of configuring its own component instances which avoids clashing
+with other kamelets or regular components (local beans). This ensures a Kamelet is a _black box_.
+
+We made using `AggregationStrategy` easier with scripting languages, which allows
+for example Kamelets to embed code as the strategy directly in the Kamelet spec file.
+
+### Health Checks
+
+Camel components can now provide custom health checks which can be automatic discovered by Camel.
```
Comment (gzurowski) [inline content/blog/2021/12/camel314-whatsnew/index.md:64]:
```suggestion
Camel components can now provide custom health checks which can be automatically discovered by Camel.
```

File: content/blog/2021/12/camel314-whatsnew/index.md [hunk 11]
```diff
@@ -0,0 +1,141 @@
...
+
+### JBang
+
+CamelJBang is used to quickly try and run Camel applications.
+
+[JBang](https://www.jbang.dev/) makes it very easy to run Java without all the ceremony.
+With CamelJBang we made it possible for anyone to quickly try Camel.
+
+The Camel Karavan Designer uses CamelJBang in its editor that allows you to visually
+builder Camel integrations and run them with ease while designing directly from the editor or web-browser.
```
Comment (gzurowski) [inline content/blog/2021/12/camel314-whatsnew/index.md:88]:
```suggestion
build Camel integrations and run them with ease while designing directly from the editor or web-browser.
```

**Review by gzurowski — APPROVED** [review]
LGTM 👍

**Comment by orpiske** [comment]
> Need to complete the kafka section
> 
> Feedback is welcome, and also fix any grammar and typos directly please.

I'll share some details about the Kafka part soon.

File: content/blog/2021/12/camel314-whatsnew/index.md [hunk 12]
```diff
@@ -0,0 +1,141 @@
...
+
+The Camel Karavan Designer uses CamelJBang in its editor that allows you to visually
+build Camel integrations and run them with ease while designing directly from the editor or web-browser.
+
+### Kafka
+
+The `camel-kafka` component has been overhauled and optimized to be more robust and have better performance.
+
+We also worked on making _resume from offset_ a general concept in Camel and provide
+more out-of-the-box strategies for `camel-kafka`.
```
Comment (orpiske) [inline content/blog/2021/12/camel314-whatsnew/index.md:95]: Here's my suggestion: 

The `camel-kafka` component has been overhauled and optimized to be more robust, have better
 [performance]((https://camel.apache.org/blog/2021/09/camel-kafka-consumer-changes) and 
 improve resource utilization. Users of the Kafka idempotent repository will also benefit from 
 some of the robustness fixes implemented during the last releases. 

Additionally, added support for async commits, we increased the test coverage, and included 
forward testing support for Kafka 3.0. 

We also worked on making _resume from offset_ a general concept in Camel and provide more 
out-of-the-box strategies for `camel-kafka`.

Comment (davsclaus) [reply content/blog/2021/12/camel314-whatsnew/index.md:95]: Thanks. To use links on the website is a pain as you cannot use absolute links and the xref stuff is "hit and miss" to figure out, so its often easier to not have any links.

**Pull request merged**